src/main/java/uk/gov/digital/ho/hocs/hocscaseworksearchindexer/ETLService.java

```diff
@@ -19,42 +22,84 @@
 import javax.persistence.EntityManager;
 import javax.persistence.PersistenceContext;
 import java.io.IOException;
-import java.util.*;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
 import java.util.concurrent.atomic.AtomicInteger;
-import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 @Slf4j
 @Service
 public class ETLService {
 
-    private int batchSize;
+    private IndexMode indexMode;
+
+    private final String prefix;
+
+    private final int batchSize;
+
+    private final boolean newIndex;
+
+    private final int batchInterval;
 
     private final CaseRepository caseRepository;
 
     private final RestHighLevelClient client;
 
     private ObjectMapper objectMapper;
 
-    private final String index;
+    private final String liveIndex;
+
+    private DateFormat dateFormat = new SimpleDateFormat("yyyyMMddhhmmss");
+
+    private final  String databaseName;
+
 
     @PersistenceContext
     protected EntityManager entityManager;
 
-    public ETLService(@Value("${batch-size}") int batchSize,
-                      CaseRepository caseRepository,
+    private final String timestamp;
+
+    public ETLService(@Value("${batch.size}") int batchSize,
                       RestHighLevelClient client,
                       ObjectMapper objectMapper,
-                      @Value("${aws.es.index-prefix}") String prefix) {
+                      @Value("${aws.es.index-prefix}") String prefix,
+                      @Value("${mode}") IndexMode indexMode,
+                      @Value("${new-index}") boolean newIndex,
+                      CaseRepository caseRepository,
+                      @Value("${batch.interval}") int batchInterval,
+                      @Value("${db.name:unset}") String databaseName) {
         this.batchSize = batchSize;
+        this.newIndex = newIndex;
         this.caseRepository = caseRepository;
         this.client = client;
+        this.prefix = prefix;
         this.objectMapper = objectMapper;
-        this.index = String.format("%s-%s", prefix, "case");
+        this.liveIndex = String.format("%s-%s", prefix, "case");
```

Comment:
`case` here likely wants to be in a variable - this would not be there going forward

Reply:
it's hardcoded in hocs-search but you're right. One for later though